### ISSUE TITLE
Recidive filter is missing my other jails' log entries

### ISSUE THREAD

**Issue by flixman** [fail2ban/fail2ban#2069]
### Environment:

_Fill out and check (`[x]`) the boxes which apply. If your Fail2Ban version is outdated, 
and you can't verify that the issue persists in the recent release, better seek support 
from the distribution you obtained Fail2Ban from_

- Fail2Ban version (including any possible distribution suffixes): 0.10.2-1
- OS, including release name/version: Linux Debian stable/testing
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:
Recidive filter does not match my systemd logs. An example of such not caught log entries is
```
   Mar 05 08:41:28 test.org fail2ban-server[11524]: fail2ban.actions        [11524]: NOTICE  [postfix-auth] Ban 94.138.214.26
```
   The failregex being used is (in recidive.conf)
```
failregex = ^(%(__prefix_line)s| %(_daemon)s%(__pid_re)s?:\s+)NOTICE\s+\[(?!%(_jailname)s\])(?:.*)\]\s+Ban\s+<HOST>\s*$
```

#### Steps to reproduce
Write this the log entry into a file (test.log) and set the failregex on a filter file, and run the fail2ban-regex tester.

#### Expected behavior
The line is catched.

#### Observed behavior
The line is missed.

#### Any additional information
I can read regular expressions, but dealing with the expanded fail2ban expressions as they are is quite difficult. Can you advise any regex debugging tool you might find useful for this purpose?

**Comment by sebres** [comment]
So you mean:
1. The logging of fail2ban configured to systemd-journal. How it is done?
```bash
fail2ban-client -d | grep -E 'logtarget|syslogsock'
```
2. Do you've configured something to log via systemd-juornal, or it was just default configuration for your debian?

3. With some wrong configuration it may log the date-time twice, once via fail2ban and once via journal.
Do you see it so?
(Notice to me. Some several cases were implemented in #1989.)

4. If case of 1 is true, your backend for jail `recidive` should be `backend = systemd` also.
```bash
fail2ban-client -d | grep -E "'add', 'recidive'"
```

5. What do you see if you try:
```bash
fail2ban-regex systemd-journal recidive
```

**Comment by sebres** [comment]
It looks like something is other in logging via syslog, the daemon-name (`fail2ban-server`) interferes with the failregexp so far (unexpected here currently).
But the logging-format are almost the same `%(name)s[%(process)d]: %(levelname)s %(message)s`.
So I assume `fail2ban-server[11524]:` will be appended via syslog resp. journal.

I'll fix the failregexp in current trunks. Thank you.
